src/rgw/rgw_sal_sfs.h

```diff
@@ -428,6 +428,21 @@ class SFStore : public Store {
   std::string get_cls_name() const { return "sfstore"; }
 };
 
+struct ScopedStoreBucketRefresher {
+
+  ScopedStoreBucketRefresher(SFSBucket &bucket)
+  : store(bucket.get_store()),
```

Comment:
I'd rather pass the store as an additional argument, and keep it as a pointer, for consistency across the codebase.

Reply:
Many types in the existing codebase are using references, so I don't see a particular problem using a reference here.
Using a pointer means the user can call it with `null`; but no, we want this to be always constructed with a bound object (and references are made for this).
Why pass the store as additional argument? We want the store to be taken from the bucket.

Reply:
I meant passing the store as a pointer. For clarity's sake, as we've been doing across the sfs codebase. I don't see the argument for taking it from the bucket to have more weight than what I'm suggesting.

Reply:
Because we want the store to be taken from the bucket reference we are passing when constructing the `ScopedStoreBucketRefresher` instance; the `ScopedStoreBucketRefresher(SFSBucket &bucket)` signature ensures this behavior.
`ScopedStoreBucketRefresher(SFSBucket &bucket, Store *store)` does not.
This because the user constructing an instance using the second signature could pass a `store` argument not taken from `bucket` argument.
The fact that `store` could be a `singleton` instance or ultimately the same object does not matter.
This is an implementation detail that could change in the future.